### PR TITLE
feat: Add ability for grouped hit requirements to track selector

### PR DESCRIPTION
**Pull request by paulgessinger**
With this PR, `TrackSelector` allows selecting tracks based on minimum hit requirements where hits are required to match a set of geometry ID masks. With this pattern, you can implement **minimum hit requirements per subdetector**, i.e. at least 3 hits in the Pixel detector, etc.

It uses `GeometryIdHierarchyMap` to implement the matching, and then combines hit counts from all matches in the map.

From python, you would set this up like:

```python
measurementCounter = acts.TrackSelector.MeasurementCounter()
# At least 2 hits in volume 3
measurementCounter.addCounter([acts.GeometryIdentifier().setVolume(3)], 2)
```

and then set it to `TrackSelectorConfig`:

```python
TrackSelectorConfig(
  pt=(1.0 * u.GeV if ttbar else 0.0, None),
  absEta=(None, 3.0),
  loc0=(-4.0 * u.mm, 4.0 * u.mm),
  nMeasurementsMin=7,
  nMeasurementsGroupMin=measurementCounter
)
```

You can also select e.g. layer IDs, where due to the OR behavior within a single map, you can model cuts like "at least 2 hits on the two innermost pixel layers".

### CODE REVIEW

**Comment by paulgessinger** [comment]
@andiwand @pbutti

**Review by andiwand — APPROVED** [review]
very nice!

**Review by andiwand — APPROVED** [review]

File: Examples/Python/python/acts/examples/reconstruction.py [hunk 1]
```diff
@@ -113,9 +113,20 @@
...
+        "loc1",
+        "time",
+        "eta",
+        "absEta",
+        "pt",
+        "phi",
+        "nMeasurementsMin",
+        "nMeasurementsGroupMin",
+    ],
     defaults=[(None, None)] * 7 + [None],
```
Comment (andiwand) [inline Examples/Python/python/acts/examples/reconstruction.py:130]:
```suggestion
    defaults=[(None, None)] * 7 + [None] * 2,
```

@paulgessinger just tested it I think this is missing to make the default construction work

Comment (paulgessinger) [reply Examples/Python/python/acts/examples/reconstruction.py]: Right, I didn't test the default construction!

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [fd39512fd88c0f9fdefdab55e4a4ee6af16905fe]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366928)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366927)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366926)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366925)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366924)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366923)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366922)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366921)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366920)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366919)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366918)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366917)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366916)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366915)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366914)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366913)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366912)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366911)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366910)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366909)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366908)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366907)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35366906)||

**Pull request merged**